lib/egg.js

```diff
@@ -351,22 +351,50 @@ class EggCore extends KoaApplication {
   }
 
   /**
-   * use co to wrap generator function to a function return promise
-   * @param  {GeneratorFunction} fn input function
-   * @return {AsyncFunction} async function return promise
+   * Convert a generator function to a promisable one.
+   *
+   * Notice: for other kinds of functions, it directly returns you what it is.
+   *
+   * @exception If `fn` isn't a function, error will be thrown out.
+   * @param  {Function} fn The inputted function.
+   * @return {AsyncFunction} An async promise-based function.
+   * @example
+   * ```javascript
+   *  const fn = function* (arg) {
+        return arg;
+      };
+      const wrapped = app.toAsyncFunction(fn);
+      wrapped(true).then((value) => console.log(value));
+   * ```
    */
   toAsyncFunction(fn) {
-    if (!is.generatorFunction(fn)) return fn;
-    fn = co.wrap(fn);
-    return async function(...args) {
-      return fn.apply(this, args);
-    };
+    if (typeof fn === 'function') {
+      // If this is a generator function, convert it with `co` to a common one.
+      if (is.generatorFunction(fn)) {
+        fn = co.wrap(fn);
+        return async function(...args) {
+          return fn.apply(this, args);
+        };
+      }
+      return fn;
+    }
+    // Other situations, throw error.
+    throw new Error('\'fn\' should be a function.');
```

Comment:
it seems a break change

Reply:
This is only for functions, so if you pass other values of types, this should be wrong. So that's the reason why I throw exceptions.

Reply:
That's right, but I'm afraid someone set a parameter which is not function works fine before, and failed after this change.

Reply:
But this seems a bug and must be fixed ASAP.
Considering what you said……, I think I can add another param called "strictMode" as an optional param to check with as a fixture way :)